### PR TITLE
Add support for generate narayana config file when the first resource added on cluster mode

### PR DESCRIPTION
**Pull request by jingshanglu**
Ref #15405.

### CODE REVIEW

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/TransactionConfigurationFileGenerator.java [hunk 1]
```diff
@@ -41,9 +41,10 @@
     /**
      * Get transaction configuration.
      *
-     * @param transactionRuleConfiguration transaction rule configuration
+     * @param originTransactionProps transaction rule props
```
Comment (LeeGuoPing) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/TransactionConfigurationFileGenerator.java:44]: The description of param `originTransactionProps  ` seems to not match very much

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/TransactionConfigurationFileGenerator.java [hunk 2]
```diff
@@ -41,9 +41,10 @@
     /**
      * Get transaction configuration.
      *
-     * @param transactionRuleConfiguration transaction rule configuration
+     * @param originTransactionProps origin transaction properties
      * @param schemaConfiguration schema configuration
+     * @param modeType mode type
      * @return transaction rule props
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/TransactionConfigurationFileGenerator.java:47]: The java doc is `Get transaction configuration`, but the return value is `transaction rule props`, please unify them

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/TransactionConfigurationFileGenerator.java [hunk 3]
```diff
@@ -41,9 +41,10 @@
...
      */
-    Properties getTransactionProps(TransactionRuleConfiguration transactionRuleConfiguration, SchemaConfiguration schemaConfiguration);
+    Optional<Properties> getTransactionProps(Properties originTransactionProps, Optional<SchemaConfiguration> schemaConfiguration, String modeType);
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/TransactionConfigurationFileGenerator.java:49]: Input param cannot permit optional value, please follow IDEA's warning to adjust it.

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-provider/shardingsphere-transaction-xa-narayana/src/main/java/org/apache/shardingsphere/transaction/xa/narayana/config/NarayanaConfigurationFileGenerator.java [hunk 4]
```diff
@@ -57,7 +56,7 @@ public void generateFile(final TransactionRule transactionRule, final InstanceCo
         String instanceId = instanceContext.getInstance().getInstanceDefinition().getInstanceId().getId();
         String recoveryId = null == instanceContext.getInstance().getXaRecoveryId() ? instanceId : instanceContext.getInstance().getXaRecoveryId();
         NarayanaConfiguration config = createDefaultConfiguration(instanceId, recoveryId);
-        if (null != transactionRule.getProps()) {
+        if (null != transactionRule.getProps() && !transactionRule.getProps().isEmpty()) {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-provider/shardingsphere-transaction-xa-narayana/src/main/java/org/apache/shardingsphere/transaction/xa/narayana/config/NarayanaConfigurationFileGenerator.java:59]: When `transactionRule.getProps()` is null?

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-provider/shardingsphere-transaction-xa-narayana/src/main/java/org/apache/shardingsphere/transaction/xa/narayana/config/NarayanaConfigurationFileGenerator.java [hunk 5]
```diff
@@ -139,21 +138,21 @@ private void appendJdbcStoreConfiguration(final String jdbcUrl, final String use
...
-    public Properties getTransactionProps(final TransactionRuleConfiguration transactionRuleConfiguration, final SchemaConfiguration schemaConfiguration) {
+    public Optional<Properties> getTransactionProps(final Properties originTransactionProps, final Optional<SchemaConfiguration> schemaConfiguration, final String modeType) {
         Properties result = new Properties();
-        if (!transactionRuleConfiguration.getProps().isEmpty()) {
-            generateUserDefinedJdbcStoreConfiguration(transactionRuleConfiguration, result);
-        } else {
-            generateDefaultJdbcStoreConfiguration(schemaConfiguration, result);
+        if (!originTransactionProps.isEmpty()) {
+            generateUserDefinedJdbcStoreConfiguration(originTransactionProps, result);
+        } else if ("Cluster".equalsIgnoreCase(modeType) && schemaConfiguration.isPresent()) {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-provider/shardingsphere-transaction-xa-narayana/src/main/java/org/apache/shardingsphere/transaction/xa/narayana/config/NarayanaConfigurationFileGenerator.java:145]: `equalsIgnoreCase` or just `equals`? Does other codes use `equalsIgnoreCase` to judge mode type?

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/TransactionConfigurationFileGenerator.java (hunk 3 above)
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/spi/TransactionConfigurationFileGenerator.java:49]: How about replace `Optional<Properties>` to empty Properties if no contents?

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Review by terrymanu — APPROVED** [review]

**Pull request merged**